### PR TITLE
Adding spark databricks integration

### LINKED ISSUE #684: [USE][DOCS] Add example how to run on data from spark
Create example of reading spark data into a dataframe and running deepchecks on it.

### PR DESCRIPTION
**Pull request by ItayGabbay**
#### Reference Issues/PRs

Closes #684 

image
image
image

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: docs/source/user-guide/integrations/spark_databricks.rst [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    trainDF, testDF = dataset.randomSplit([0.8, 0.2], seed=42)
+
+Defining deepchecks Dataset
+---------------------------
+
+.. code-block:: python
+
+    from deepchecks.tabular import Dataset
+
+    pd_train = trainDF.toPandas()
```
Comment (noamzbr) [inline docs/source/user-guide/integrations/spark_databricks.rst:42]: perhaps note here that sampling is recommended and is OK logically as most checks sample anyway?

File: docs/source/user-guide/integrations/spark_databricks.rst [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    from pyspark.ml.classification import LogisticRegression
+    from pyspark.ml import Pipeline
+
+    lr = LogisticRegression(featuresCol="features", labelCol="label")
+
+    # Define the pipeline based on the stages created in previous steps.
+    pipeline = Pipeline(stages=[stringIndexer, encoder, labelToIndex, vecAssembler, lr])
+
+    # Define the pipeline model.
```
Comment (noamzbr) [inline docs/source/user-guide/integrations/spark_databricks.rst:112]:
```suggestion
    # Fit the pipeline model.
```

File: docs/source/user-guide/integrations/spark_databricks.rst [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    pipelineModel = pipeline.fit(trainDF)
+
+Writing a Model Wrapper
+~~~~~~~~~~~~~~~~~~~~~~~
+We will write a wrapper to our model, that will implement the required API for deepchecks according the the
```
Comment (noamzbr) [inline docs/source/user-guide/integrations/spark_databricks.rst:117]: Personally I would mention what these requirements are here.

File: docs/source/user-guide/integrations/spark_databricks.rst [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        def predict_proba(self, X: np.ndarray) -> np.ndarray:
+            df=spark.createDataFrame(X)
+            preds = self.model.transform(df).select('prediction').collect()
+            return np.array(preds).reshape(-1, 2)
+
+        @property
+        def feature_importances_(self):
+            return np.array([1/14] * 14)
+
+.. note::
```
Comment (noamzbr) [inline docs/source/user-guide/integrations/spark_databricks.rst:146]: Maybe note here that if the user has some FI series of is own making, he can pass it to the run() method of the suites.

**Review by noamzbr — APPROVED** [review]
Approved with a few comments / suggestions.

File: docs/source/user-guide/integrations/spark_databricks.rst [hunk 5]
```diff
@@ -0,0 +1,170 @@
+Spark & Databrick
```
Comment (shir22) [inline docs/source/user-guide/integrations/spark_databricks.rst:1]:
```suggestion
Spark & Databricks
```

File: docs/source/user-guide/integrations/spark_databricks.rst [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+        @property
+        def feature_importances_(self):
+            return np.array([1/14] * 14)
+
+.. note::
+    The wrapper here considers that all features are equally important. This is not a valid assumption for
+    real models, but is done here for simplicity.
+
+Running the Full Suite
```
Comment (shir22) [inline docs/source/user-guide/integrations/spark_databricks.rst:150]: Lets replace this with a model evaluation suite and (/or) a train-test validation suite
(to "publicize" less the full suite)

**Review by shir22 — APPROVED** [review]
Approved with comments

File: docs/source/user-guide/integrations/spark_databricks.rst (hunk 4 above)
Comment (ItayGabbay) [reply docs/source/user-guide/integrations/spark_databricks.rst:146]: Added a link to the new feature importances guide instead :)

**Pull request merged**